libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -503,6 +509,7 @@ AP_GPS_UBLOX::read(void)
 // Private Methods /////////////////////////////////////////////////////////////
 void AP_GPS_UBLOX::log_mon_hw(void)
 {
+#ifndef HAL_NO_LOGGING
```

Comment:
This double-negative is kind of painful.

Reply:
it matches all the other HAL_NO_* defines we use when the default is for something to be enabled